index.js

```diff
@@ -1,88 +1,239 @@
+        typescript: true,
+        vue: true,
+
+        // We are more conservative on curly braces
+        // <https://github.com/antfu/eslint-config/blob/0cd12cc90d2100798a5d8f5d51b34753b7be7f70/src/configs/stylistic.ts#L54-L63>
+        lessOpinionated: true
+      },
+      {
+        rules: {
+          'max-statements-per-line': 'off'
         }
-      ],
-      alphabetize: {
-        order: 'asc',
-        caseInsensitive: false
       }
-    }],
-    'new-cap': ['error', {
-      newIsCap: false,
-      capIsNew: false,
-      properties: true
-    }],
-    'n/prefer-global/process': 'off',
+    )
+      .override('antfu/javascript/rules', {
+        rules: {
+          'no-cond-assign': 'off',
+          'new-cap': [
+            'error',
+            {
+              newIsCap: false,
+              capIsNew: false,
+              properties: true
+            }
+          ]
+        }
+      })
+      .override('antfu/stylistic/rules', {
+        rules: {
+          'style/arrow-parens': ['error', 'always'],
+          'style/brace-style': [
+            'error',
+            '1tbs',
+            {
+              allowSingleLine: true
+            }
+          ],
+          'style/comma-dangle': ['error', 'never']
+        }
+      })
+      .override('antfu/typescript/rules', {
+        rules: {
+          'ts/consistent-type-definitions': 'off',
+          'ts/consistent-type-imports': [
+            'error',
+            {
+              prefer: 'type-imports',
+              disallowTypeAnnotations: false,
+              fixStyle: 'inline-type-imports'
+            }
+          ],
+          'ts/no-empty-interface': 'off',
+          // Can't enable this rule because it conflicts with our preference of inline type imports.
+          'ts/no-import-type-side-effects': 'off',
+          'ts/no-unused-vars': [
+            'error',
+            {
+              argsIgnorePattern: '^_',
+              destructuredArrayIgnorePattern: '^_'
+            }
+          ],
+          'ts/no-use-before-define': 'off'
+        }
+      })
+      .override('antfu/perfectionist/setup', {
+        settings: {
+          perfectionist: {
+            // Perfectionist uses `localeCompare` to sort characters by default,
+            // which is different from the default `sort` behavior we used to have.
+            // So we need to set a custom alphabet to sort characters by their char code.
+            // Note: this freezes the @eslint/config-inspector page when expanding the `perfectionist` settings,
+            // as it generates too many characters.
+            alphabet: Alphabet.generateRecommendedAlphabet()
+              .sortByCharCodeAt()
+              // Nested imports come before the parent imports,
+              // i.e. `@/components/foo/nested.vue` before `@/components/foo.vue`
+              // TODO: do we really need this?
+              .placeCharacterBefore({ characterBefore: '/', characterAfter: '.' })
```

Comment:
I think I would like the opposite of that lol 😅 foo.vue should come before foo/nested.vue. Though we don't do such things.

Reply:
Can't remember where we've had that kind of sorting… Anyway, that's such a rare case that I can't even find an example now, I think it would be fine to remove this override here.